public/js/dashboard.js

```diff
@@ -61,50 +61,61 @@ requirejs([
         $container.removeClass('invisible');
     }
 
+    // delegates for kernel message handling, keyed by message type
+    // all functions receive the entire response message and a reference to an
+    // outputArea model associated with the code / widget that made the initial
+    // request associated with the received response
+    var messageHandlers = {
+        clear_output: function(msg, outputArea) {
+            outputArea.clear(msg.content.wait);
+        },
+        stream: function(msg, outputArea) {
+            var output = {};
+            output.outputType = OutputType.Stream;
+            output.text = msg.content.text;
+            switch(msg.content.name) {
+                case "stderr":
+                  // show stderr in console, not in the dashboard itself
+                  console.error(msg.content.name, msg.content.text);
+                  break;
+                case "stdout":
+                  output.name = StreamName.StdOut;
+                  outputArea.add(output);
+                  break;
+                default:
+                  throw new Error('Unrecognized stream type ' + msg.content.name);
+            }
+        },
+        display_data: function(msg, outputArea) {
+            var output = {};
+            output.outputType = OutputType.DisplayData;
+            output.data = msg.content.data;
+            output.metadata = msg.content.metadata;
+            outputArea.add(output);
+        },
+        execute_result: function(msg, outputArea) {
+            var output = {};
+            output.outputType = OutputType.ExecuteResult;
+            output.data = msg.content.data;
+            output.metadata = msg.content.metadata;
+            output.execution_count = msg.content.execution_count;
+            outputArea.add(output);
+        },
+        error: function(msg, outputArea) {
+            // show tracebacks in the console, not on the page
+            var traceback = msg.content.traceback.join('\n');
+            console.error(msg.content.ename, msg.content.evalue, traceback);
```

Comment:
In the future, will we want to show something in the page, so user knows that a given section of dashboard failed? Otherwise, user may not know to look in console for error details. Maybe a simple red icon?